src/Blt/Plugin/Commands/SwsCommands.php

```diff
@@ -366,4 +367,108 @@ protected function getSiteAliases() {
+   *
+   * @command sws:post-code-deploy
+   *
+   * @aliases sws:post-code-update
+   */
+  public function postCodeDeployUpdate($target_env, $deployed_tag, $options = ['no-slack' => FALSE]) {
+    $sites = $this->getConfigValue('multisites');
+    $parallel_executions = (int) getenv('UPDATE_PARALLEL_PROCESSES') ?: 10;
+
+    $site_chunks = array_chunk($sites, ceil(count($sites) / $parallel_executions));
+    $commands = [];
+    foreach ($site_chunks as $sites) {
+      $commands[] = $this->blt()
+        ->arg('sws:update-sites')
+        ->arg(implode(',', $sites))
+        ->getCommand();
+    }
+    file_put_contents(sys_get_temp_dir() . '/update-report.txt', '');
+    $this->taskExec(implode(" &\n", $commands) . PHP_EOL . 'wait')->run();
+    $report = array_filter(explode("\n", file_get_contents(sys_get_temp_dir() . '/update-report.txt')));
+
+    $success = [];
+    $failed = [];
+    foreach ($report as $line) {
+      [$site, $status] = explode(':', $line);
+      if ((int) $status) {
+        $success[] = $site;
+      }
+      else {
+        $failed[] = $site;
+      }
+    }
+    unlink(sys_get_temp_dir() . '/update-report.txt');
+
+    $this->yell(sprintf('Updated %s sites successfully.', count($success)), 100);
+    $slack_url = $options['no-slack'] ? FALSE : getenv('SLACK_NOTIFICATION_URL');
+
+    if ($failed) {
+      $this->yell(sprintf("Update failed for the following sites:\n%s", implode("\n", $failed)), 100, 'red');
+
+      if ($slack_url) {
+        $count = count($failed);
+        $this->sendSlackNotification($slack_url, "A new deployment has been made to *$target_env* using *$deployed_tag*.\n\n*$count* sites failed to update.");
+      }
+      throw new \Exception('Failed update');
+    }
+
+    if ($slack_url) {
+      $this->sendSlackNotification($slack_url, "A new deployment has been made to *$target_env* using *$deployed_tag*.");
+    }
+  }
+
+  /**
+   * Send out a slack notification.
+   *
+   * @param string $message
+   *   Slack message.
+   */
+  protected function sendSlackNotification(string $slack_webhook, string $message) {
+    $client = new Client();
+    $client->post($slack_webhook, [
+      'form_params' => [
+        'payload' => json_encode([
+          'username' => 'Acquia Cloud',
+          'text' => $message,
+          'icon_emoji' => 'information_source',
+        ]),
+      ],
+    ]);
+  }
+
+  /**
+   * Run db updates and config imports to a list of sites.
+   *
+   * @command sws:update-sites
+   *
+   * @var string $sites
+   *   Comma delimited list of sites to update.
+   */
+  public function updateSites($sites, $options = ['partial-config-import' => FALSE]) {
+    $sites = explode(',', $sites);
+    foreach ($sites as $site_name) {
+      $this->switchSiteContext($site_name);
+      $task = $this->taskDrush();
+      if (!$options['partial-config-import']) {
+        $task->drush('updatedb')
+          ->drush('config:import')
+          ->option('partial');
```

Comment:
Drush `deploy` also triggers a drush deploy hook `drush deploy:hook`. I know we don't use it, but maybe we'd want to add it to the end here just for parallellism (is that a word)?

Reply:
Hmm. I like that.

Reply:
I could be optimized to run after all the db updates and config imports, but i don't think it's worth the heavy lift on that.